functions/supplier/debezium-supplier/src/test/java/org/springframework/cloud/fn/supplier/debezium/custom/DebeziumEngineBuilderAutoConfigurationTest.java

```diff
@@ -57,8 +57,8 @@
  */
 @Tag("integration")
 @Testcontainers
-public class DebeziumEngineAutoConfigurationTest {
-	private static final Log logger = LogFactory.getLog(DebeziumEngineAutoConfigurationTest.class);
+public class DebeziumEngineBuilderAutoConfigurationTest {
```

Comment:
I think naming this "AutoConfigurationTest" is not accurate as it is an integration test of the functionality. 

I think we should keep this test, rename it `DebeziumEngineBuilderIntegrationTests`, create another `DebeziumEngineBuilderAutoConfgurationTests` that verifies autoconfig features such as the following:

- `@ConditionalOnMissingBean` beans can be customized (aka `custom<xyz>IsRespected()`) for the : offsetCommitPolicy, debeziumClock, completionCallback, connectorCallback ([example](https://github.com/spring-projects/spring-pulsar/blob/12e557c79ad7c923d538a044dfe67297ead167da/spring-pulsar-spring-boot-autoconfigure/src/test/java/org/springframework/pulsar/autoconfigure/PulsarAutoConfigurationTests.java#L131))
- the `debeziumEngineBuilder` is configured using the passed in properties
- and also important is the `OnDebeziumConnectorCondition` where we can easily use the `ApplicationContextRunner.withFilteredClassLoader` to verify the condition

On one hand it might seem like we are testing simple things that should just work. Yes, but it is more about putting that in place so when someone decides to refactor later and then the "custom xyz" is no longer respected and user finds the bug. Similar to how some unit tests seem a waste at the beginning, but they end up paying off. I prefer to treat the auto-config in this same way. You will find the same treatment in the boot repo for the auto-configs that they support.